tools/convertUsageStatsLogFile.php

```diff
@@ -0,0 +1,570 @@
+            ]
+        ];
+        $application = Application::get();
+        $applicationName = $application->getName();
+        switch ($applicationName) {
+            case 'ojs2':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'article/download'],
+                    Application::ASSOC_TYPE_SUBMISSION => [
+                        'article/view'],
+                    Application::ASSOC_TYPE_ISSUE => [
+                        'issue/view'],
+                    Application::ASSOC_TYPE_ISSUE_GALLEY => [
+                        'issue/download']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'index';
+                break;
+            case 'omp':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'catalog/download'],
+                    Application::ASSOC_TYPE_MONOGRAPH => [
+                        'catalog/book'],
+                    Application::ASSOC_TYPE_SERIES => [
+                        'catalog/series']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'catalog/index';
+                break;
+            case 'ops':
+                $pageAndOp = $pageAndOp + [
+                    Application::ASSOC_TYPE_SUBMISSION_FILE => [
+                        'preprint/download'],
+                    Application::ASSOC_TYPE_SUBMISSION => [
+                        'preprint/view']
+                ];
+                $pageAndOp[Application::getContextAssocType()][] = 'index';
+                break;
+        }
+        return $pageAndOp;
+    }
+
+    /**
+     * Set assoc type and IDs from the passed page, operation and arguments.
+     */
+    protected function setAssoc(int $assocType, string $op, array $args, array &$newEntry): void
+    {
+        switch ($assocType) {
+            case Application::ASSOC_TYPE_SUBMISSION_FILE:
+                if (!isset($args[0]) || !isset($args[1])) {
+                    echo "Missing URL parameter.\n";
+                    break;
+                }
+                $submissionId = (int) $args[0];
+                $submissionExists = Repo::submission()->exists($submissionId, $newEntry['contextId']);
+                if (!$submissionExists) {
+                    echo "Submission with the ID {$submissionId} does not exist in the context (journal, press or server) with the ID {$newEntry['contextId']}.\n";
+                    break;
+                }
+
+                $representationId = (int) $args[1];
+                $galley = Repo::galley()->get($representationId);
+                if (!$galley) {
+                    echo "Represantation (galley or publication format) with the ID {$representationId} does not exist.\n";
+                    break;
+                }
+
+                if (Application::get()->getName() == 'omp') {
```

Comment:
I'd prefer if this was a `switch` statement explicitly mentioning each valid case, rather than trusting existing differences between applications. For example,
```php
case 'ojs2':
case 'ops':
    break; // noop
case 'omp':
    ...run the conditonal code...
    break;
default: throw new Exception('Unrecognized application name!');
```
This will help pinpoint areas of the code needing attention if e.g. we add another application to the suite in the future.